### PR TITLE
Enforce style rules for C# 8/9

### PR DESCRIPTION
**Pull request by RoosterDragon**
Following on from #20765, and in the same vein as other PRs to turn on style rules, enable all the rules that were disabled due to previously not being on C# 8/9.

It is easier to review rules by commit. We can alter the rule options or not turn on rules if they are not satisfactory in the style they are applying.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -598,11 +598,6 @@ dotnet_diagnostic.SA1633.severity = none # FileMustHaveHeader
 dotnet_diagnostic.SA1642.severity = none # ConstructorSummaryDocumentationShouldBeginWithStandardText
 dotnet_diagnostic.SA1649.severity = none # FileNameMustMatchTypeName
 
-# Requires C# 8/9 - TODO Consider enabling
-dotnet_diagnostic.SA1141.severity = none # UseTupleSyntax
-dotnet_diagnostic.SA1316.severity = none # TupleElementNamesShouldUseCorrectCasing
-dotnet_diagnostic.SA1414.severity = none # TupleTypesInSignaturesShouldHaveElementNames
```
Comment (abcdefg30) [inline .editorconfig:604]: I thought the idea was to list all available options, even if we set the severity to the default?

Comment (RoosterDragon) [reply .editorconfig:604]: That's the setup for the IDExxxx rules. For the stylecop ones we currently have those enabled en-masse and the turn off individual rules.

**Comment by pchote** [comment]
Most of the rule changes here look good or neutral to me. I don't really like IDE0063 because IMO there is value in having the scope of those objects being explicitly specifified. Also not convinced about IDE0078 because I feel that C# has gone way overboard with inventing a new distinct syntax for pattern matching.

**Comment by RoosterDragon** [comment]
IDE0063 and IDE0078 no longer enabled.

**Review by pchote — APPROVED** [review]

**Comment by penev92** [comment]
Fwiw the using thing felt pretty good after the initial shock. And the pattern matching is hit and miss, having some pretty good and some not-so-good instances. (The two rules that are now gone.)

**Comment by penev92** [comment]
It is weird that we use
`IDE0083 Use pattern matching ('not' operator)`
but not
`IDE0078 Use pattern matching`
but okay.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5d9381062f6945e8d0cb0e5803d20e9ada54c0be...9a57fd44cd82f93e2fdd2bf2222395d6d3569391)

**Pull request merged**